### PR TITLE
upgrade spdlog to 1.8.5 and above

### PR DESCRIPTION
**Pull request by rongou**
Fallout from https://github.com/rapidsai/rmm/pull/658.

### CODE REVIEW

File: conda/recipes/versions.yaml
Comment (kkraus14) [inline conda/recipes/versions.yaml:136]: Should we pin this specifically to `1.8.2` or at least pin to `>=1.8.2,<1.9.0a0` or `>=1.8.2,<2.0.0a0`?

Comment (rongou) [reply conda/recipes/versions.yaml:136]: Hopefully `spdlog` would be stable enough going forward to not break our builds, but I'd be fine either way.

Comment (kkraus14) [reply conda/recipes/versions.yaml:136]: I'm definitely not comfortable allowing it to roll to new major versions, but I'd defer to @harrism @jrhemstad or others for if they have thoughts here on how they want to handle this dependency

Comment (harrism) [reply conda/recipes/versions.yaml:136]: Spdlog already broke on cuda once where it worked in an older version. I don't think cUDA stability is their priority. So probably should lock it to specific releases unless we contribute tests and CI to spdlog.

**Comment by rongou** [comment]
rerun tests

File: conda/recipes/versions.yaml
Comment (rongou) [reply conda/recipes/versions.yaml:136]: @kkraus14 looking at their release history, I don't think there are planned major changes between different point releases.

@harrism the break was caused by `fmt`, right? Our CIs would definitely catch any breaking changes, and then we can restrict the versions. Would that be enough?

I'm hesitant to add unnecessary version restrictions as they also affect downstream projects. For example, if someone wants to use a newer a version of `spdlog` and they also depend on rmm/cudf, they would be out of luck if we lock the version.

Comment (harrism) [reply conda/recipes/versions.yaml:136]: That's true. OK.

**Comment by rongou** [comment]
rerun tests

**Comment by mike-wendt** [comment]
rerun tests

File: conda/recipes/versions.yaml
Comment (mike-wendt) [reply conda/recipes/versions.yaml:136]: @rongou I agree with @kkraus14 that we should restrict it to at least `>=1.8.2,<1.9.0a0` or `>=1.8.2,<2.0.0a0`

You've recently reviewed their docs, I'll let you decided which restriction we should use. I understand your comment, but due to the issues we've experienced in the past and the issues with keeping working conda builds over a release we cannot leave this unpinned

**Review by mike-wendt — CHANGES_REQUESTED** [review]

File: conda/recipes/versions.yaml
Comment (mike-wendt) [reply conda/recipes/versions.yaml:136]: The same pinning should reflect in `rmm` when the decision is made

**Comment by mike-wendt** [comment]
Blocked until `runtime` images are fixed

**Comment by mike-wendt** [comment]
rerun tests

**Comment by mike-wendt** [comment]
@rongou there are also conda conflicts in the `gpuCI/build/env-pkg-nightly` tests from the previous run. I expect them to happen again so we need to find out what else uses `spdlog` and which version they depend on as the current pinning does not seem to work

**Comment by mike-wendt** [comment]
rerun tests

**Comment by kkraus14** [comment]
@mike-wendt conflicts are from `spdlog 1.8.2` requiring `libgcc-ng >=9.3.0`. We may need to update our build-stack more generally as new packages are only going to be produced with this runtime version moving forward in conda-forge.

**Comment by mike-wendt** [comment]
rerun tests

File: conda/recipes/versions.yaml
Comment (rongou) [reply conda/recipes/versions.yaml]: Added pinning.

**Comment by mike-wendt** [comment]
With the revert of #199 in #200 we'll need to create a workaround by publishing this one off pkg. Holding this PR until we can get a gcc 7.5 version published

**Review by mike-wendt — APPROVED** [review]
Approved, but unable to merge

**Comment by rongou** [comment]
Is it ok to merge this now?

**Comment by kkraus14** [comment]
> Is it ok to merge this now?

No, this needs to be passing: https://github.com/rapidsai/rmm/pull/658 as well as cudf

**Review by kkraus14 — CHANGES_REQUESTED** [review]
RMM / cuDF need to be addressed before merging

**Comment by rongou** [comment]
I believe https://github.com/rapidsai/rmm/pull/658 can only pass after this goes in.

**Comment by kkraus14** [comment]
> I believe [rapidsai/rmm#658](https://github.com/rapidsai/rmm/pull/658) can only pass after this goes in.

You can manually upgrade `spdlog` in the `ci/gpu` script and make sure everything solves and passes as expected, then we can merge this, then we can revert that change and merge the RMM PR.

**Comment by mike-wendt** [comment]
This can be tested and merged after #202 is merged

**Review by mike-wendt — APPROVED** [review]
Waiting on new testing after #202 is merged

**Comment by mike-wendt** [comment]
Will wait for merging until new gpuCI images are out and the changes in #202 have propagated

**Comment by mike-wendt** [comment]
> RMM / cuDF need to be addressed before merging

@kkraus14 do we still need to address this? or can you re-review?

**Comment by kkraus14** [comment]
> > RMM / cuDF need to be addressed before merging
> 
> @kkraus14 do we still need to address this? or can you re-review?

We still haven't tested libcudf as far as I know. We need to open a PR bumping the version in the build there to make sure it doesn't cause issues (it likely shouldn't).

**Comment by raydouglass** [comment]
@kkraus14 Now that we've moved to a 9.3.0 build stack, we should be able to merge this right? What are the next steps regarding rmm & cudf?

**Comment by kkraus14** [comment]
> @kkraus14 Now that we've moved to a 9.3.0 build stack, we should be able to merge this right? What are the next steps regarding rmm & cudf?

We need to open PRs in at least RMM and cuDF making sure the update works as expected (not sure about other libraries, likely Blazing as well).

**Comment by mike-wendt** [comment]
@kkraus14 who should take on the PR testing to get this over the line?

**Comment by kkraus14** [comment]
> @kkraus14 who should take on the PR testing to get this over the line?

We should probably test similarly on cuDF and BlazingSQL at the minimum before we move forward. Will try to ping folks to see if they can replicate this testing.

**Comment by rongou** [comment]
@kkraus14 can this be merged now? The RMM tests are passing https://github.com/rapidsai/rmm/pull/658.

**Comment by kkraus14** [comment]
> @kkraus14 can this be merged now? The RMM tests are passing [rapidsai/rmm#658](https://github.com/rapidsai/rmm/pull/658).

Would still like to test cudf before merging. I'll open a PR bumping spdlog there for testing purposes.

**Comment by kkraus14** [comment]
> > @kkraus14 can this be merged now? The RMM tests are passing [rapidsai/rmm#658](https://github.com/rapidsai/rmm/pull/658).
> 
> Would still like to test cudf before merging. I'll open a PR bumping spdlog there for testing purposes.

Opened PR here to test: https://github.com/rapidsai/cudf/pull/8067

**Comment by kkraus14** [comment]
FYI -- Pinged BlazingSQL folks since they use spdlog where they're confirming that 1.8.5 is okay for them and then we'll merge.

**Comment by raydouglass** [comment]
> FYI -- Pinged BlazingSQL folks since they use spdlog where they're confirming that 1.8.5 is okay for them and then we'll merge.

@kkraus14 Any update on this?

**Comment by kkraus14** [comment]
> > FYI -- Pinged BlazingSQL folks since they use spdlog where they're confirming that 1.8.5 is okay for them and then we'll merge.
> 
> @kkraus14 Any update on this?

PR is being thrown up to confirm this doesn't cause issues for them now. Hopefully should be straightforward.

**Comment by rongou** [comment]
@kkraus14 have you heard back from them?

**Comment by kkraus14** [comment]
> @kkraus14 have you heard back from them?

Yes, apologies, this slipped through the cracks on my end. We should be good to go to proceed here.

**Comment by rongou** [comment]
@kkraus14 can you approve this? Thanks!

**Review by kkraus14 — APPROVED** [review]

**Comment by rongou** [comment]
rerun tests

**Comment by rongou** [comment]
rerun tests

**Comment by ajschmidt8** [comment]
@rongou, @kkraus14 just want to confirm whether or not this PR is okay to be merged. the comments seem to indicate that it is, but the `DO NOT MERGE` label is still applied. Should we remove that label now?

**Comment by kkraus14** [comment]
Should be good to go, apologies.

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**
